### ISSUE TITLE
Ability to run azure functions will be supported only starting with .NET 9?

### ISSUE THREAD

**Issue by Depact** [captainsafia/AspireAzureFunctionsSample#3]
Is it possible to have .net 8 project with azure functions in aspire using this approach?

**Comment by captainsafia** [comment]
@Depact This depends on what you mean by .NET 9. Your Functions projects that target the `net8.0` TFM will still work as you can see from the sample app:

https://github.com/captainsafia/AspireAzureFunctionsSample/blob/24de24750462e6e67c1234e96d3ab53699837734/AzureFunctionsTest/AzureFunctionsTest.Functions/AzureFunctionsTest.Functions.csproj#L3

However, you will need to use the 9.0 versions of Aspire and the .NET 9 SDK. Both of these will still permit you to target the .NET 8 runtime in your actual Functions application.

**Comment by Depact** [comment]
@captainsafia thanks for the response. I noticed that projects targets .Net 8 when cloned this project and tried convert it to 8. No signs of luck on this yet. Wasn't sure that it's right repo to ask can it even be done. 
9 is not an option because there is no stable and probably wouldn't be for around 2 months. That's unacceptable for enterprise usage.
So I'm in need of something to run it on 8 or ditch aspire approach and use something else.

**Comment by captainsafia** [comment]
> 9 is not an option because there is no stable and probably wouldn't be for around 2 months. That's unacceptable for enterprise usage.

Yep, this repo is intended to be a sample app for people who are interested in trialing the preview support and seeing if it meets their needs. It's also the sample app we are using to validate the integration. The Azure Functions support in Aspire is currently in progress and we anticipate releasing a V1 as part of .NET 9/Aspire 9.

**Comment by Depact** [comment]
Thanks for the detailed response. As far as I see you've given azure on aspire quite an effort and probably one of the most if not most competent people in this direction. But my answer wasn't answered. Is there any way to run azure functions exclusively on LTS versions of packages and sdk?
Before going neck deep into forking till I get it running on LTS SDK, is there anything against doing it or anything that can come handy on such venture?
But first I probably will give a shot to idea of using preprocessor directives to enable azure functions only if 9 used.

**Comment by Depact** [comment]
Okay defyning custom `NET9` constant, using preprocessor directives and csproj ItemGroup condition did the trick for me. For my project that targets .NET8 it solves the issue. Thanks for giving me the idea. Feel free to close this item. Or I will close later
`NET9_0_OR_GREATER` not worked.

**Comment by captainsafia** [comment]
@Depact Can you share the changes you made just so we're on the same page?

You shouldn't need to change any TargetFrameworks in your project but you will need to use 9.0.x versions of the Aspire packages.

**Comment by Depact** [comment]
@captainsafia I will not be able to share exact changes for couple of days, but basically what I did.
I had some azure function and non azure function projects added in aspire project. All of them .NET 8.
I defined NET9 constant in csproj. 
Added two itemGroups, one with .NET 9 versions of packages and one with .NET 8 and tied to NET9 constant I created. So when constant is enabled .NET 9 dependencies used.
used `#if NET9` in code around connecting azure functions.
This way everyone with .NET 8, including pipeline, can build solution. But if .NET 9 is used azure functions also available for hosting using aspire.
This approach was inspired by your words that project targets .NET 8 but uses .NET 9 packages and mention that .NET 9 SDK is backwards compatible. This way actual .NET 8 SDK can be used to build it. But also .NET 9 too.

**Comment by captainsafia** [comment]
@Depact Thanks for the detailed description! Your approach looks correct.

Let me know if you have any feedback/questions on the Aspire + Azure Functions integration in .NET 9/

**Comment by Depact** [comment]
@captainsafia for now my biggest pain is that logs buttons moved to pop up at some moment. I want to see logs when I'm using Aspire.
Other things:
- All my projects start with same word, and final generated names are quite big. So they take plenty of space in column, with handy part clipping out.
- No clue what `Traces` and `Metrics` tabs do yet. Would suggest putting in the middle of nothingness link to docs + explanation what supposed to be here. Very relevant when opening those tabs and have no idea what is going on there.
- I appreciate that Aspire starts all components when launching `AppHost` but I'm yet to find out clean solution to start only some of them. This probably rare but quite important for me use case. Kinda handy when trying to debug only some Azure Functions.
- This chunk of code triggers decent code analysis tools and PR reviewers. 
```
namespace Microsoft.Extensions.Hosting;

// Adds common .NET Aspire services: service discovery, resilience, health checks, and OpenTelemetry.
// This project should be referenced by each service project in your solution.
// To learn more about using this project, see https://aka.ms/dotnet/aspire/service-defaults
public static class Extensions
```
Likely there is no other ways, but very evoking questions from anyone onboarded on this tech.
- I used Azure TFS agents to build my aspire project in pipeline and it's failed returning odd not very useful message, because it was pretty much nothing than code and link to docs iirc. After two days of investigating each and every line of code I found out that for some reason one agent pool was returning 8.0.100 and another agent pool was returning 8.0.401 of .NET SDK, same with Aspire workload. But figuring out that it's because of obscure issue with TFS getting different versions. Message with something other than error code would've been really helpful. 
- Not found yet any example or doc mentioning that installing aspire workload should be before `dotnet restore` in pipeline script, figured out only because in some runs it returned me error.

I'm not on PC with Aspire currently so my list quite fuzzy on detail. If needed can provide more detailed info in Friday for any point.
